v1.38.0 breaks due to missing include

I'm trying to [upgrade google-cloud-cpp to grpc v1.38.0](https://github.com/googleapis/google-cloud-cpp/pull/6621) and the build breaks because this file doesn't include `<vector>`

The errors from [our build log](https://storage.googleapis.com/cloud-cpp-community-publiclogs/logs/google-cloud-cpp/6621/ae762e2afbff8375f2cb74af2908bdfd3f257ae9/fedora-33-bazel-targets/log-0d856e2f-62be-429e-86b2-9bc4e21a8ae5.txt):

```
Step #3: In file included from external/com_github_grpc_grpc/src/core/lib/iomgr/error.h:33,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/iomgr/closure.h:33,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/channel/handshaker.h:33,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/security/security_connector/security_connector.h:28,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/security/security_connector/ssl/ssl_security_connector.h:26,
Step #3:                  from external/com_github_grpc_grpc/src/core/lib/security/security_connector/ssl/ssl_security_connector.cc:21:
Step #3: external/com_github_grpc_grpc/src/core/lib/gprpp/status_helper.h:139:6: error: 'vector' in namespace 'std' does not name a template type
Step #3:   139 | std::vector<absl::Status> StatusGetChildren(absl::Status status)
Step #3:       |      ^~~~~~
Step #3: external/com_github_grpc_grpc/src/core/lib/gprpp/status_helper.h:25:1: note: 'std::vector' is defined in header '<vector>'; did you forget to '#include <vector>'?
Step #3:    24 | #include "absl/status/status.h"
Step #3:   +++ |+#include <vector>
Step #3:    25 | #include "absl/time/time.h"
Step #3: ERROR: /h/.cache/bazel/_bazel_root/9bd401995a189c62294f86ba79716c4c/external/com_github_grpc_grpc/BUILD:223
```

_Originally posted by @devjgm in https://github.com/grpc/grpc/pull/25901#r638120250_

It looks like this file is also missing an include of `absl/types/optional`

--- a/src/core/lib/gprpp/status_helper.h
+++ b/src/core/lib/gprpp/status_helper.h
@@ -21,8 +21,11 @@
 
 #include <grpc/support/port_platform.h>
 
+#include <vector>
+
 #include "absl/status/status.h"
 #include "absl/time/time.h"
+#include "absl/types/optional.h"
 
 #include "src/core/lib/gprpp/debug_location.h"
 

